crates/p2p/src/main_loop.rs

```diff
@@ -399,152 +412,182 @@ impl MainLoop {
+                tracing::debug!(%peer, %request_id, "Sync request sent");
 
-                    self.event_sender
-                        .send(Event::InboundBodiesSyncRequest {
-                            from: peer,
-                            request,
-                            channel,
-                        })
-                        .await
-                        .expect("Event receiver not to be dropped");
-                }
-                request_response::Message::Response {
+                let _ = self
+                    .pending_sync_requests
+                    .bodies
+                    .remove(&request_id)
+                    .expect("Block sync request still to be pending")
+                    .send(Ok(channel));
+            }
+            SwarmEvent::Behaviour(behaviour::Event::TransactionsSync(
+                p2p_stream::Event::InboundRequest {
                     request_id,
-                    response,
-                } => {
-                    let _ = self
-                        .pending_sync_requests
-                        .bodies
-                        .remove(&request_id)
-                        .expect("Block sync request still to be pending")
-                        .send(Ok(response));
-                }
-            },
+                    request,
+                    peer,
+                    channel,
+                },
+            )) => {
+                tracing::debug!(?request, %peer, %request_id, "Received sync request");
+
+                self.event_sender
+                    .send(Event::InboundTransactionsSyncRequest {
+                        from: peer,
+                        request,
+                        channel,
+                    })
+                    .await
+                    .expect("Event receiver not to be dropped");
+            }
             SwarmEvent::Behaviour(behaviour::Event::TransactionsSync(
-                request_response::Event::Message { message, peer },
-            )) => match message {
-                request_response::Message::Request {
-                    request, channel, ..
-                } => {
-                    tracing::debug!(?request, %peer, "Received sync request");
+                p2p_stream::Event::OutboundRequestSentAwaitingResponses {
+                    request_id,
+                    peer,
+                    channel,
+                },
+            )) => {
+                tracing::debug!(%peer, %request_id, "Sync request sent");
 
-                    self.event_sender
-                        .send(Event::InboundTransactionsSyncRequest {
-                            from: peer,
-                            request,
-                            channel,
-                        })
-                        .await
-                        .expect("Event receiver not to be dropped");
-                }
-                request_response::Message::Response {
+                let _ = self
```

Comment:
Do we really not care about the result here? If so, this could use a comment ;)

Reply:
There's plenty of places where the same comment would have to be added ("if the sending fails the client is dropped and we cannot do much about it").